datadog-checks-base/datadog_checks/checks/win/winpdh.py

```diff
@@ -59,27 +75,32 @@ def __init__(self, class_name, counter_name, log, instance_name = None, machine_
                     self.logger.error("%s doesn't seem to be a multi-instance counter, but asked for specific instance %s" % (
                         class_name, instance_name
                     ))
-                    return
+                    raise AttributeError("%s is not a multi-instance counter" % class_name)
                 if instance_name not in instances:
                     self.logger.error("%s is not a counter instance in %s" % (
                         instance_name, class_name
                     ))
-                    return
+                    raise AttributeError("%s is not an instance of %s" % (instance_name, class_name))
             path = self._make_counter_path(machine_name, counter_name, instance_name, counters)
-            try:
-                self.logger.debug("Path: %s\n" % unicode(path))
-            except: # noqa: E722
-                # some unicode characters are not translatable here.  Don't fail just
-                # because we couldn't log
-                self.logger.debug("Failed to log path")
-                pass
-            try:
-                self.counterdict[SINGLE_INSTANCE_KEY] = win32pdh.AddCounter(self.hq, path)
-            except: # noqa: E722
-                self.logger.fatal("Failed to create counter.  No instances of %s\%s" % (
-                    self._class_name, counter_name))
-                raise
-            self._is_single_instance = True
+            if path is None or len(path) == 0:
+                self.logger.warning("Empty path returned")
+            else:
+                try:
+                    self.logger.debug("Path: %s\n" % unicode(path))
```

Comment:
If you want you could try `unicode(path, errors='ignore')`.

Reply:
The problem is in `logger.debug`; it sometimes throws exceptions and I really don't want to stop processing b/c of a log error.